core/store/src/main/java/org/locationtech/geowave/core/store/base/BaseDataStore.java

```diff
@@ -232,13 +249,27 @@ public DataStatisticsStore getStatisticsStore() {
 		return internalQuery(
 				query.getQueryConstraints(),
 				queryOptions,
-				delete);
+				delete,
+				filterDuplicates);
 	}
 
 	protected <T> CloseableIterator<T> internalQuery(
 			final QueryConstraints constraints,
 			final BaseQueryOptions queryOptions,
-			final boolean delete ) {
+			final boolean delete,
+			final boolean filterDuplicates ) {
+
+		// Note: filterDuplicates has a different effect depending if delete is
+		// true or not.
+		// If delete is false and filterDuplicates is true, the query will
+		// return
+		// just the entries from the tables corresponding to the specified range
+		// If delete is false and filterDuplicates is false, the query will
+		// return all the entries (duplicates included)
+		// if delete is true and filterDuplicates is true, the query will filter
+		// duplicates, but the duplicate deletion callback WILL NOT BE added
+		// if delete is true and filterDuplicates is false, the query will
```

Comment:
I can tighten up the language here but the underlying problem is this: If you perform a deletion query on a partial range (e.g. 1970-1971, full range is 1970-1975, dupes are 1972-1975), it will only delete the partial range (1970-1971) regardless if that filter is null or not. This is what I meant by "duplicates still get filtered", I suppose I should say "they won't get deleted". According to the logic in `BaseConstraintsQuery` the dedupe filter is only applied client side as well, hence why it has no effect when delete is called and why I created the duplicate deletion scan callback. I updated the comment, but let me know if you think there is a better path or way to implement this.